channeldb/channel.go

```diff
@@ -1537,7 +1539,9 @@ func (c *OpenChannel) AdvanceCommitChainTail(fwdPkg *FwdPkg) error {
 		if err != nil {
 			return err
 		}
-		err = putChanCommitment(chanBucket, &newCommit.Commitment, false)
+		err = putChanCommitment(
+			chanBucket, &newCommit.Commitment, false, false,
```

Comment:
i think it might be better to read the channel status and compute `isRestoredChan` before calling here an above. this way we will fail if we try to put commitments on restored chans

Reply:
See comment above. In light of that, do you mean to error out all together if any of the mutation calls are executed for a restarted channel?

Reply:
yeah i think that'd be a good safety net, that way we catch any misuses of restored channels, see other comment

Reply:
Added a safety net, also extended the tests to exercise this.